packages/Interface/src/DTK_C_API.h

```diff
@@ -0,0 +1,206 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+#ifndef DTK_C_API_H
+#define DTK_C_API_H
+
+#include <DataTransferKit_config.hpp>
+
+#include <DTK_Types.h>
+
+#include <stddef.h> // for size_t
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*! Return current DTK version. */
+extern const char *DTK_version();
+
+/*! Get current repository hash.
+ *
+ * Returns an error string if it is not a GIT repository
+ */
+extern const char *DTK_git_commit_hash();
+
+typedef void DTK_Handle;
+
+/*! Create a DTK handle. */
+extern DTK_Handle *DTK_create();
+
+/*! Destroy a DTK handle */
+extern void DTK_destroy( DTK_Handle **handle );
+
+/*! Initialize DTK */
+extern void DTK_initialize( int *argc, char ***argv );
+
+/*! Check whether DTK is initialized */
+extern int DTK_is_initialized();
+
+/*! Finalize DTK */
+extern void DTK_finalize();
+
+typedef enum {
+    DTK_NODE_LIST_SIZE_FUNCTION,
+    DTK_NODE_LIST_DATA_FUNCTION,
+    DTK_BOUNDING_VOLUME_LIST_SIZE_FUNCTION,
+    DTK_BOUNDING_VOLUME_LIST_DATA_FUNCTION,
+    DTK_POLYHEDRON_LIST_SIZE_FUNCTION,
+    DTK_POLYHEDRON_LIST_DATA_FUNCTION,
+    DTK_CELL_LIST_SIZE_FUNCTION,
+    DTK_CELL_LIST_DATA_FUNCTION,
+    DTK_MIXED_TOPOLOGY_CELL_LIST_SIZE_FUNCTION,
+    DTK_MIXED_TOPOLOGY_CELL_LIST_DATA_FUNCTION,
+    DTK_BOUNDARY_SIZE_FUNCTION,
+    DTK_BOUNDARY_DATA_FUNCTION,
+    DTK_DOF_MAP_SIZE_FUNCTION,
+    DTK_DOF_MAP_DATA_FUNCTION,
+    DTK_MIXED_TOPOLOGY_DOF_MAP_SIZE_FUNCTION,
+    DTK_MIXED_TOPOLOGY_DOF_MAP_DATA_FUNCTION,
+    DTK_FIELD_SIZE_FUNCTION,
+    DTK_PULL_FIELD_DATA_FUNCTION,
+    DTK_PUSH_FIELD_DATA_FUNCTION,
+    DTK_EVALUATE_FIELD_FUNCTION
+} DTK_FUNCTION_TYPE;
+
+/*! Register a function with DTK. */
+extern void DTK_set_function( DTK_Handle *handle, DTK_FUNCTION_TYPE type,
```

Comment:
I'm not sure I like having one blanket set function. I think this is ultimately because of the existence of the named function. We should talk about named functions and see if we cant get rid of them.